### PR TITLE
Add 'My Tales' filter back to Browse view

### LINKED ISSUE #511: Implement My Tales filter
With the completion of the [browse refactor](https://github.com/whole-tale/dashboard/pull/478) it's no longer possible to filter on only my tales (My Tales button is disabled).  Clicking the "My Tales" button should display my tales with parity with current UI (access level write/admin) and "All/Public Tales" should show everything (access level read).

### PR DESCRIPTION
**Pull request by bodom0015**
### Problem
After refactoring the Browse view, the "My Tales" filter has been disabled.

Fixes #511 

### Approach
Re-enable the "My Tales" filter.

NOTE: This PR does not alter the method by which the tales are searched or filtered.

### How to Test
Prerequisites: At least one Tale accessible not owned by you

1. Checkout and run this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the Browse view
    * By default, you should be shown "All Tales" (API automatically filters GET /tales to only those accessible to the current user)
4. On the top-left, click the "My Tales" filter button
    * You should see the contents of the view change to only show you items you have created
    * NOTE: this could be adjusted to filter on `_accessLevel`, if desired

### CODE REVIEW

**Comment by Xarthisius** [comment]
LGTM! Minor nitpick: when user has no Tales text in "My Tales" tab says: `There are no tales to display for applied filter: all`. The following patch fixed it for me:

```
diff --git a/app/components/ui/tale-browser/component.js b/app/components/ui/tale-browser/component.js
index 0fb8ec7..058465e 100644
--- a/app/components/ui/tale-browser/component.js
+++ b/app/components/ui/tale-browser/component.js
@@ -8,7 +8,6 @@ import $ from 'jquery';
 const O = EmberObject.create.bind(EmberObject);
 
 export default Component.extend({
-  currentTab: 'all',
   store: service(),
   userAuth: service('user-auth'),
   apiCall: service('api-call'),
diff --git a/app/components/ui/tale-browser/template.hbs b/app/components/ui/tale-browser/template.hbs
index f7f60e7..f2a126d 100644
--- a/app/components/ui/tale-browser/template.hbs
+++ b/app/components/ui/tale-browser/template.hbs
@@ -149,7 +149,7 @@
     
     {{#unless modelsInView.length}}
         <div class="ui center aligned segment" style="color: black;">
-            There are no tales to display for applied filter: {{currentTab}}
+            There are no tales to display for applied filter: {{filter}}
         </div>
     {{/unless}}
     {{#if loadingTales}}

```

**Comment by Xarthisius** [comment]
Noticed something more serious: deleting a Tale in "My Tales" tab renders dashboard somewhat unresponsive if you go back to the "All Tales" (e.g. there's no view button when I hover over a Tale).

I see the following backtrace in the console:

```app.js:22 TypeError: Cannot read property 'name' of undefined
    at Class.<anonymous> (component.js:13)
    at ComputedPropertyPrototype.get (ember-metal.js:4069)
    at get (ember-metal.js:3439)
    at RootPropertyReference.compute (ember-glimmer.js:326)
    at RootPropertyReference.value (ember-glimmer.js:203)
    at valueOf (runtime.js:4236)
    at Array.map (<anonymous>)
    at CapturedPositionalArguments.value (runtime.js:4216)
    at SimpleHelperReference.compute (ember-glimmer.js:506)
    at SimpleHelperReference.value (ember-glimmer.js:203)
```
`component.js` from above stack is `app/components/ui/delete-modal/component.js`

### Steps to reproduce
1. Create Ligo Tale.
2. Copy and Launch Ligo
3. Stop "Copy" Ligo Tale
4. Navigate to Browse > My Tales and delete copied Ligo Tale
5. (optionally) Navigate to Browse > All Tales and hover over Ligo Tale

**Comment by bodom0015** [comment]
Thanks, @Xarthisius!

I think I have addressed these issues.. I'm not sure how suddenly the `delete-modal` was acting up, but I've adjusted it to use only the modelName (since this could either be `name` or `title`, better to just let the caller decide which string to pass in).

This appears to have gotten rid of that gross stack trace, and the "View" buttons appear to stay now.

Let me know if this clears up the issue for you as well!

**Comment by Xarthisius** [comment]
@bodom0015 after updating to 723700c8c6165b585b76e43115c8f1a69ab9e3ab and login I get:

bug

here's the console output:

```
Attributes updated
wholetale-c9f8dd9c9489daf30eddc6445e66f441.js:38 TypeError: Cannot read property 'syscall' of null
    at e.evaluate (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1146)
    at e.evaluateSyscall (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1387)
    at e.evaluateInner (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1381)
    at e.evaluateOuter (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1381)
    at e.next (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1518)
    at e.next (vendor-468b71615cf72303033dfaa7aa6e61b3.js:1519)
    at e.render (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2081)
    at e.runInTransaction (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2289)
    at t.e._renderRoots (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2104)
    at t.e._renderRootsTransaction (vendor-468b71615cf72303033dfaa7aa6e61b3.js:2109)
```

**Comment by bodom0015** [comment]
@Xarthisius very weird... that stack trace is definitely not from our code (`vendor.js`), and we shouldn't be using `syscall` directly anywhere since I'm not even sure what that is or does...

Have you tried in incognito? As much as I hate the "it works on my machine" argument, I'm not sure what that error means or where it could have come from

**Comment by Xarthisius** [comment]
> @Xarthisius very weird... that stack trace is definitely not from our code (`vendor.js`), and we shouldn't be using `syscall` directly anywhere since I'm not even sure what that is or does...
> 
> Have you tried in incognito? As much as I hate the "it works on my machine" argument, I'm not sure what that error means or where it could have come from

I nuked `node_modules` and `bower_components`, installed everything again, tried incognito in chrome and even installed firefox. Same behavior.

**Comment by Xarthisius** [comment]
@bodom0015 removing
```
{{ui/copy-on-launch-modal taleToCopy=taleToCopy modelsInView=modelsInView}}
```
from `app/components/ui/tale-browser/template.hbs` "fixes" it.

File: app/components/ui/tale-browser/template.hbs [hunk 1]
```diff
@@ -424,25 +423,4 @@
...
-      <i class="remove icon"></i>
-      Cancel
-    </div>
-    <div class="ui positive right labeled icon button" {{action 'submitCopyAndLaunch' taleToCopy}}>
-      Copy and Run
-      <i class="checkmark icon"></i>
-    </div>
-  </div>
-</div>
+{{ui/copy-on-launch-modal taleToCopy=taleToCopy modelsInView=modelsInView}}
```
Comment (Xarthisius) [inline app/components/ui/tale-browser/template.hbs]: /me waves at @bodom0015

**Comment by Xarthisius** [comment]
> @bodom0015 removing
> 
> ```
> {{ui/copy-on-launch-modal taleToCopy=taleToCopy modelsInView=modelsInView}}
> ```
> 
> from `app/components/ui/tale-browser/template.hbs` "fixes" it.

Resolved by merging https://github.com/whole-tale/dashboard/pull/516

**Pull request merged**